src/metrics.rs

```diff
@@ -52,24 +52,28 @@
 //! EXPOSE 9090
 //! ```
 
-use log::error;
-use warp::{Filter, Rejection, Reply};
-
-pub use lazy_static::lazy_static;
-
-pub use prometheus::*;
+use std::net::SocketAddr;
 
 use crate::error::DshError;
+use bytes::Bytes;
+use http_body_util::{BodyExt, Full};
+use hyper::body::Incoming;
+use hyper::server::conn::http1;
+use hyper::service::service_fn;
+use hyper::{header, Method, Request, Response, StatusCode};
+use hyper_util::rt::TokioIo;
+pub use lazy_static::lazy_static;
+use log::error;
+pub use prometheus::register_int_counter;
+use prometheus::Encoder;
+pub use prometheus::IntCounter;
+use tokio::net::TcpListener;
 
-/// Encode metrics to a string (UTF8)
-pub fn metrics_to_string() -> std::result::Result<String, DshError> {
-    let encoder = prometheus::TextEncoder::new();
+type GenericError = Box<dyn std::error::Error + Send + Sync>;
+type Result<T> = std::result::Result<T, GenericError>;
```

Comment:
Please add possible errors to DshError in error.rs and use DshError as returned error.
This remark can be ignored if we get rid of this error in the public API